ncl/ncl_entries/specx_specxy_anal.ipynb

```diff
@@ -0,0 +1,2260 @@
+      "text/plain": [
+       "<Figure size 640x480 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "soi = soi_darwin.DSOI\n",
+    "soi['time'] = datetime_index\n",
+    "\n",
+    "soi.plot();"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Detrend\n",
+    "\n",
+    "Here we will do a simple detrending by removing the mean with [`scipy.signal.detrend()`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.detrend.html).\n",
+    "\n",
+    "If you wanted to perform a linear least-squares fit, change `type='linear'`.\n",
+    "\n",
+    "The `scipy.signal.periodogram()` function does have a detrending option, but because we are using an unsupported tapering window, and because detrending must be done before tapering, here we demonstrate how to separate out these steps."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "soi_detrended = scipy.signal.detrend(soi, type='constant')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Tapering\n",
+    "\n",
+    "According to the code in [NCL's `taper.f`](https://github.com/NCAR/ncl/blob/edd46f5426286146313b31bd6826e3d3c360c0dc/ni/src/lib/nfpfort/taper.f), the tapering performed in the NCL example is split-cosine-bell tapering.\n",
+    "\n",
+    "NCL also has a [`taper_n()`](https://www.ncl.ucar.edu/Document/Functions/Built-in/taper_n.shtml) function that is preferable for performing tapering on specific dimensions, and is reported to be faster.\n",
+    "\n",
+    "It is unclear which tapering function is used within the NCL spectral analysis functions.\n",
```

Comment:
Isn't this the tapering function for spectral analysis functions? https://github.com/NCAR/ncl/blob/edd46f5426286146313b31bd6826e3d3c360c0dc/ni/src/lib/nfpfort/specx_dp.f#L702

It is interesting that there seem to be a bunch of them in the codebase.

Reply:
Would I point to this function as NCL's `specx_dp.f.DTAPERX` or do I drop the f?

Reply:
I think I would refer to it as the DTAPERX subroutine and link to the line(s) in the code.